### DISCUSSION TITLE
objectType level resolver

### DISCUSSION THREAD

**Discussion by georgekrax** [graphql-nexus/nexus#1045] [Q&A]
I would like to create a field like this:
```js
t.string("formattedLocation", {
  description: "Get the location of the #beach_bar formatted",
  resolve: async ({ id, ...hey }, __, { prisma }): Promise<string> => {
    const location = await prisma.beachBarLocation.findUnique({
      where: { beachBarId: +id },
      include: { region: true, city: true, country: true },
    });
    if (!location) return "";

      let formattedLocation: string[] = [];
      if (location.region) formattedLocation = [...formattedLocation, location.region.name];
      if (location.city) formattedLocation = [...formattedLocation, location.city.name]
      if (location.country) formattedLocation = [...formattedLocation, location.country.alpha2Code];
      return formattedLocation.join(", ");
  },
});
```
So, `hey.location` is not already filled.
But the parent element does not have the `location` field already filled. What can I do? Should I create a local-only field in the Apollo Client and not on the server?

Every help is appreciated!

**Comment by ericyd** [comment]
I don't see any technical problems with the code you posted. What problem are you running into?

I don't understand this line
>But the parent element does not have the location field already filled. What can I do?

This feels more like a question regarding your specific app, rather than one about Nexus. Is this a problem with implementing a field resolver, or a problem with your domain modeling (which is probably not fitting for this board)?

**Reply by georgekrax** [reply]
As I see no other response, I should probably close this.
